docs/arch/adr-009-storage-migration.md

```diff
@@ -0,0 +1,74 @@
+# ADR 9: OpenEBS to OpenEBS + Rook Migration
+
+## Context
+
+Single node installs should not use Rook, as doing so adds complexity, increases resource usage, and invites disk corruption.
+Multi node installs very often require distributed storage, best provided by Rook.
+
+Many customers will move from a single-node install to a multi-node install as circumstances change.
+This migration should be as simple as possible.
+Vendors generally should also not need to specify multiple installer specs for single and multi-node installs.
+
+## Decision
+
+Provide a migration path from OpenEBS to OpenEBS + Rook, triggered by the presence of 3+ nodes.
+
+## Solution
+
+A new field, "minimum node count", will be added to the rook config. (minimum value 2, requires openebs localpv to be present)
+When it is set, the rook operator will be installed on single node installs, but no ceph cluster will be created.
+
+An example spec is provided below:
+
+```yaml
+spec:
+  rook:
+    version: "1.10.x"
+    storageClassName: "distributed"
+    minimumNodeCount: 3
+  openebs:
+    version: "3.4.x"
+    isLocalPVEnabled: true
+    localPVStorageClassName: "local"
+```
+
+When installed on a single node, two storageclasses will be available - "scaling" and "local". 
+The default storage class will be "scaling", and both storage classes will be backed by openebs localpv.
+
+When the third node is added, a Ceph cluster will be created by the EKCO operator.
+When that Ceph cluster becomes healthy (with at least 3 replicas), the "distributed" storageclass will be created using rook-ceph.
+However, a migration will not begin until one of three things occurs:
+
+1. The user approves the migration in kotsadm
+2. The user runs `install.sh` on a primary node, and accepts a prompt to migrate storage
+3. The user runs the `migrate-multinode-storage` command in `tasks.sh` from a primary node
```

Comment:
I'd prefer that this be a go command directly so that it is reusable. If the only interface we have for the end user is tasks.sh then the shell should only invoke a go command and nothing else.

Reply:
Agreed that the logic in tasks.sh should be minimal-to-nonexistent